### PR TITLE
v0.4.0

### PR DESCRIPTION
**Pull request by Paebbels**
# New Features
* New test cases to increase code coverage.
  (Especially, each Argument class is now tested standalone.)
* Added VSCode settings file.
* Added example to documentation landing page.
* Added new documentation content and diagrams.
* Added doc-strings to almost every module, class or method.
* Cross referenced doc-strings and normal documentation pages.  
  Added `see also` sections.
* Added `todo` boxes for undocumented sections with item lists.
* Added consumer section to documentation landing page.
* Started to document `Program` and `Executable` in standalone chapters.
* Added key-value-pair flags (`-gDebug=True`).

# Changes
* Tested many variants on how a Program is instantiated.
* Reworked diagrams.
* Extracted some information from diagrams into tables for better visualization/comparison.
* Added checks for MacOS.
* Exchanged Jinja template for `autoapi`.
* Modified options for `autodoc` to include inherited, private and special members.
* Massive renaming and class name cleanups to simplify usage of Argument and Flag classes.
  * Split classes of similar style into new Python modules.
* Updated README.
* Updated News section in documentation landing page.

# Bug Fixes
* Test case fixed on MinGW64 by installing `msys/git`.
* Fixed wrong package name in `.gitignore`.

--------------------
Related PRs:
* #7
* https://github.com/pyTooling/Actions/pull/39

### CODE REVIEW

File: doc/Arguments.rst [hunk 1]
```diff
@@ -1,83 +1,262 @@
+.. _ARG:
+
 Arguments
 #########
 
+.. todo:: Nameing convertion
```
Comment (umarcor) [inline doc/Arguments.rst:6]: "Naming".

File: doc/Arguments.rst [hunk 2]
```diff
@@ -1,83 +1,262 @@
...
+   * Named arguments |rarr| Flag
+   * Character prefixes |rarr| Short, Long, Windows
+
+
+.. _ARG:Overview:
+
+Overview
+********
+
+.. mermaid::
```
Comment (umarcor) [inline doc/Arguments.rst:18]: Might be worth having this in a file and include/import it?

File: doc/Program.rst [hunk 3]
```diff
@@ -1,4 +1,77 @@
 Program
 #######
 
+The :class:`~pyTooling.CLIAbstraction.Program` represents an executable command line program. It offers an interface to
+define and enabled command line arguments.
```
Comment (umarcor) [inline doc/Program.rst:5]: `s/enabled/enable/`

File: doc/Tutorial.rst [hunk 4]
```diff
@@ -1,4 +1,9 @@
 Tutorial
 ########
 
+.. todo:: Needs to be written.
 
+   * How to select Program vs. Executable.
+   * How to define a first argument as a nested class derived from existing classes.
+   * How to chose from existing argument classes.
```
Comment (umarcor) [inline doc/Tutorial.rst:8]: `s/chose/choose/`

File: doc/index.rst [hunk 5]
```diff
@@ -44,24 +44,103 @@ Main Goals
...
 *********
 
-*tbd*
+* Wrap command line interfaces of EDA tools (Electronic Design Automation) in Python classes.
+
+
+Examples
+********
+
+The following example implements a portion of the ``git`` program and its ``--version`` argument.
```
Comment (umarcor) [inline doc/index.rst:65]: This block shows subcommand "commit", not argument "--version".

File: pyTooling/CLIAbstraction/NamedKeyValuePair.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+# Unless required by applicable law or agreed to in writing, software                                                  #
+# distributed under the License is distributed on an "AS IS" BASIS,                                                    #
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.                                             #
+# See the License for the specific language governing permissions and                                                  #
+# limitations under the License.                                                                                       #
+#                                                                                                                      #
+# SPDX-License-Identifier: Apache-2.0                                                                                  #
+# ==================================================================================================================== #
+#
+"""Flag arguments represent simple boolean values by being present or absent.
```
Comment (umarcor) [inline pyTooling/CLIAbstraction/NamedKeyValuePair.py:32]: This docstring seems copied from `Flag.py`?

File: doc/Arguments.rst (hunk 2 above)
Comment (Paebbels) [reply doc/Arguments.rst:18]: Maybe later. I'm not satisfied yet.

File: doc/index.rst (hunk 5 above)
Comment (Paebbels) [reply doc/index.rst]: Fixed in fbd47bfc.

File: doc/Tutorial.rst (hunk 4 above)
Comment (Paebbels) [reply doc/Tutorial.rst]: Fixed in fbd47bfc.

File: doc/Program.rst (hunk 3 above)
Comment (Paebbels) [reply doc/Program.rst]: Fixed in fbd47bfc.

File: doc/Arguments.rst (hunk 1 above)
Comment (Paebbels) [reply doc/Arguments.rst]: Fixed in fbd47bfc.

File: pyTooling/CLIAbstraction/NamedKeyValuePair.py (hunk 6 above)
Comment (Paebbels) [reply pyTooling/CLIAbstraction/NamedKeyValuePair.py:32]: Oh. I missed to rework key-value-pairs.

I'll rework them and write test cases for them too ...

Comment (Paebbels) [reply pyTooling/CLIAbstraction/NamedKeyValuePair.py]: It's now implemented in e8b5028.

**Comment by Paebbels** [comment]
@umarcor packaging problem is solved by running CI with an experimental branch from pyTooling/Actions.

The matching PR is this: https://github.com/pyTooling/Actions/pull/39

**Comment by Paebbels** [comment]
README is now also up-to-date.

**Comment by Paebbels** [comment]
CI is now green. See https://github.com/pyTooling/pyTooling.CLIAbstraction/actions/runs/1921649165

**Pull request merged**